draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,545 @@ author:
 
-Proxy servers MUST NOT include the "DNS-SVCB-Params" response header field if the
-corresponding request did not include a "DNS-SVCB-Keys". Servers MAY include
-specific SvcParamKey values that were not requested. Specifically, servers SHOULD include
-the "mandatory" parameter if present, which would be presented as "p0", along with any
-parameters that are defined as mandatory for that record.
+Each item in the list has the following parameters:
 
-As an example, assume that the server received the following "svc.example.com" SVCB records:
+* t: The DNS RR type of the DNS query name, which MUST be specified
+  as an sf-integer (so 1 for A, 28 for AAAA, and 5 for CNAME).
+
+* ttl: The TTL of the address or CNAME record SHOULD be included as a
+  parameter with the key "ttl", whose value MUST be an sf-integer.
+  If the value of of intermediate records are unavailable,
+  the ttl parameter SHOULD be included on just the last list
+  entry, containing the minimum TTL value across the CNAMEs.
+
+* o: The owner name of the DNS record MAY be included.
+  This this parameter is specified, its value MUST be an sf-string
+  whose value is the owner name of the record.
+
+
+
+### Example Proxy-DNS-Used response
+
+As an example, assume that the proxy server resolved the
+authority hostname of "svc.example.com" as follows:
 
 ~~~ diagram
-   svc.example.com. 3600 IN HTTPS 1 svc2.example.com. alpn=h2,h3 echconfig="123..."
-   svc.example.com. 3600 IN HTTPS 2 . alpn=h2 echconfig="abc..."
+   svc.example.com.  7200 IN CNAME  svc.example.net.
+   svc.example.net.  1800 IN CNAME  svc2.example.net.
+   svc2.example.net.   60 IN A      192.0.2.74
+   svc2.example.net.   60 IN AAAA   2001:db8::75
+   svc2.example.net.   60 IN AAAA   2001:db8::76
 ~~~
 
-A successful CONNECT response would include the following headers, if the client requested both
-"alpn" and "echconfig":
+If the proxy connected to 2001:db8::75 then the successful CONNECT
+response would include the following header:
 
 ~~~ example
 HEADERS
 :method = CONNECT
 :status = 200
-dns-svcb-params = "svc2.example.com.";priority=1;ttl=3600;p1=:aDIsaDM=:;p5=:MTIzLi4u:,
-                  "svc.example.com.";priority=2;ttl=3600;p1=:aDI=:;p5=:YWJjLi4u:
+proxy-dns-used = "svc.example.net.";ttl=7200;t=5;o="svc.example.com.",
+                 "svc2.example.net.";ttl=1800;t=5;o="svc.example.net.",
+                 "2001:db8::75";ttl=60;t=28;o="svc2.example.net."
 ~~~
 
+
+*TODO*: Do we need the "o=" parameter or is it redundant?
+        If we include it, should it be a MUST or MAY?
+        
+*TODO*: Do we need to cover DNAME as well? 
+
+*TODO*: Should a future version be able to include NS record
+        and DNSSEC information?
+
+
+# Proxy Behavior
+
+Proxy servers MUST NOT take action based on SVCB records.
+In particular, the ipv4hint and ipv6hint SvcParams MUST NOT
+be used by proxies for making connections.
+
+*TODO*: Discuss if this is too restrictive
```

Comment:
Hm, why wouldn't they be able to do this? This does seem a bit restrictive.

Reply:
It's a topic for discussion.  Some reasons:
- This does change proxy semantics in normative ways.  (ie, it is changing how the proxy is deciding on what to connect to).  There may be security risks/exposures here, for example.  
- This could cause load-balancing issues as those may be smaller sets than A/AAAA and the TTLs may be longer.
This should probably get a first-class issue to discuss.  (I'm thinking a WG would also want to discuss this as a topic.)